### PR TITLE
[SYCL][Bindless] Refactor existing bindless aspects for missing L0 bindless functionality

### PR DESCRIPTION
**Pull request by DBDuncan**
Add device aspects query if a device supports image arrays, unique addressing per dimension and support for sampling 1D usm images.

Remove export semaphore, export image memory and fetch 3D usm image aspects as they are not supported on any backend.

Update Level Zero and CUDA aspect results to reflect current support status with latest drivers.

### CODE REVIEW

**Comment by DBDuncan** [comment]
Corresponding UR PR: https://github.com/oneapi-src/unified-runtime/pull/1795

**Review by isaacault — APPROVED** [review]

**Comment by DBDuncan** [comment]
Friendly ping @intel/dpcpp-tools-reviewers @intel/unified-runtime-reviewers @intel/llvm-reviewers-runtime @againull

**Review by againull — APPROVED** [review]

**Comment by DBDuncan** [comment]
Friendly ping @intel/dpcpp-tools-reviewers

**Comment by DBDuncan** [comment]
@intel/dpcpp-tools-reviewers Review please.

**Comment by DBDuncan** [comment]
@sarnex @intel/dpcpp-tools-reviewers Please review. Need approval as soon as you can.

**Review by sarnex — APPROVED** [review]
lgtm

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -99,13 +99,13 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-  #     Merge pull request #1803 from cppchedy/chedy/new-ur-copy-api
-  #     [CUDA][L0][Bindless] update design of urBindlessImagesImageCopyExp
-  set(UNIFIED_RUNTIME_TAG d2b086a7170145f255e8702f512f6157b047cbe5)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/DBDuncan/unified-runtime.git")
+  # commit 40902ecace6d16500c79e5787314e3e393e70379
+  # Author: Duncan Brawley <duncan.brawley@codeplay.com>
+  # Date:   Wed May 29 13:04:15 2024 +0100
+  #     [Bindless][Exp] Update device_info enum elements and update Level Zero
+  #     and CUDA backend device_info supported features
+  set(UNIFIED_RUNTIME_TAG 40902ecace6d16500c79e5787314e3e393e70379)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:108]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit f11caf98857bb22800d09aedec8bd607643ab67b
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Tue Jul 23 15:28:10 2024 +0100
  #     Merge pull request #1795 from DBDuncan/duncan/bindless-aspects
  #     [Bindless][Exp] Update device_info enum elements and update Level Zero and CUDA backend device_info supported features
  set(UNIFIED_RUNTIME_TAG f11caf98857bb22800d09aedec8bd607643ab67b)
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1795 has been merged, please update the UR repo/tag as suggested and pull in the latest changes from the sycl branch.

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (DBDuncan) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: Done.

**Review by kbenzie — APPROVED** [review]
UR LGTM

**Comment by DBDuncan** [comment]
@intel/llvm-gatekeepers LGTM

**Pull request merged**